src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -678,11 +678,23 @@ protected override void Shutdown()
             }
             else
             {
-                VariableDetails result =
-                    await editorSession.DebugService.EvaluateExpression(
+                VariableDetailsBase result;
+                if (evaluateParams.Context.Equals("watch", StringComparison.OrdinalIgnoreCase))
+                {
+                    // If we evaluate an expression from a watch context then the server goes into a deadlock.
+                    // So instead, we retrieve the value from already evaluated variables.
+                    result = editorSession.DebugService.GetVariableFromExpression(
```

Comment:
Yep.  I think it's better if we can figure out a way to deal with the root cause here, I'll chat with Kapil about it when I get to the office later.